test/test_nn.py

```diff
@@ -494,6 +494,15 @@ def constructor_args(self):
     def extra_args(self):
         return self._get_arg('extra_args', False)
 
+class TestLongTensor(TestCase):
```

Comment:
was the comment about depthwise, etc handled?

Reply:
depthwise is supported only for CUDA. I added transpose2d and dilated2d

Reply:
Added conv3d, transpose3d, dilated3d

Reply:
Instead of doing one-off tests like this, how about adding this as an option to the normal NN Module / Criterion tests?  There is already some use of this, see `check_eval`.

Reply:
What is the benefit to move them to CriterionTest?

Reply:
it will check way more combinations of parameters.

Reply:
The problem with this approach is that initially Tensor is generated as random float and it's impossible to compare it with the long tensor.

Reply:
Fixed tests